### PR TITLE
feat(harbor): validate partition task names against task_source at build time

### PR DESCRIPTION
**Pull request by shehabyasser-scale**
Stacked on #9; companion to #16 (defense in depth for the same live incident). A Mode B partition with bare TB2 names (`break-filter-js-from-html` instead of `terminal-bench/break-filter-js-from-html`) compiled fine, then zeroed an entire optimization trial at eval time after burning its budget. #16 makes the eval-time failure loud; this PR catches it at build time where it costs $0.

The compiler enumerates the registry `task_source` (best-effort, via harbor's `DatasetConfig`) and fails the build naming the unknown tasks + canonical examples. Offline compiles warn and continue; `VERO_SKIP_TASK_NAME_CHECK=1` bypasses; local `inner_task` sources are exempt.

3 tests (unknown fails, canonical passes, offline warns). 11 pass.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

<h3>Greptile Summary</h3>

This PR adds a build-time guard that validates partition task names against the registry's canonical `<org>/<name>` form, catching the class of bug where bare names compiled silently and zeroed an entire optimization trial at eval time. The implementation is best-effort: offline or harbor-unavailable compiles warn and continue; `VERO_SKIP_TASK_NAME_CHECK=1` bypasses; `inner_task` local sources are exempt.

- `_resolve_task_source_names` enumerates task names via `DatasetConfig.get_task_configs()`, handling both sync and async-caller contexts via a `ThreadPoolExecutor` workaround for running-loop environments.
- `_validate_partition_names` raises a descriptive `ValueError` on unknown names, treats both `None` and empty-set returns as warn + continue, and is inserted into `compile_task` before dataset registration so a bad config costs nothing.
- Five tests cover the unknown/canonical/offline/empty-source/bypass-env-var paths, addressing all three issues flagged in the prior review round.

<details><summary><h3>Confidence Score: 5/5</h3></summary>

Safe to merge — the new validation is best-effort and fails open when enumeration is unavailable, so it cannot block legitimate offline builds or introduce regressions on existing compile paths.

All three issues from the prior review round are addressed: the nested-event-loop bypass, the confusing empty-source error, and the missing bypass-env-var test. The validation is inserted before any filesystem side effects on the build output, and the async thread-worker approach for running-loop contexts is correct. No logic errors or unsafe paths were found.

No files require special attention.
</details>

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| vero/src/vero/harbor/build/compiler.py | Adds _resolve_task_source_names and _validate_partition_names; hooks validation into compile_task before dataset registration. Handles async-context edge case via ThreadPoolExecutor. Logic is sound and well-guarded. |
| vero/tests/test_harbor_build.py | Adds 5-test TestPartitionNameValidation class covering all key branches (unknown fails, canonical passes, offline warns, empty-source warns, env-var bypass). Responds to all three prior review threads. |

</details>

<details><summary><h3>Flowchart</h3></summary>

<a href="#gh-light-mode-only">

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A[compile_task Mode B] --> B{task_source set and no inner_task?}
    B -- No --> E[build_harbor_dataset + register]
    B -- Yes --> C{VERO_SKIP_TASK_NAME_CHECK?}
    C -- Yes --> E
    C -- No --> D[_resolve_task_source_names]
    D --> F{names returned?}
    F -- None or empty set --> G[logger.warning: skipping check]
    G --> E
    F -- non-empty set --> H{unknown names in partition?}
    H -- No --> E
    H -- Yes --> I[raise ValueError with sample canonical names]
    D --> J{Running event loop?}
    J -- No --> K[asyncio.run coro]
    J -- Yes --> L[ThreadPoolExecutor asyncio.run in worker thread]
    K --> M[return set of names]
    L --> M
```

</a>
<a href="#gh-dark-mode-only">

```mermaid
%%{init: {'theme': 'base', 'themeVariables': {"darkMode": true, "background": "#0d1117", "primaryColor": "#21262d", "primaryTextColor": "#e6edf3", "primaryBorderColor": "#8b949e", "lineColor": "#8b949e", "textColor": "#e6edf3", "edgeLabelBackground": "#161b22", "actorBkg": "#21262d", "actorBorder": "#8b949e", "actorTextColor": "#e6edf3", "actorLineColor": "#8b949e", "signalColor": "#8b949e", "signalTextColor": "#e6edf3", "noteBkgColor": "#373320", "noteBorderColor": "#d4a72c", "noteTextColor": "#f0e6c0", "labelBoxBkgColor": "#21262d", "labelBoxBorderColor": "#8b949e", "labelTextColor": "#e6edf3", "loopTextColor": "#e6edf3", "activationBkgColor": "#30363d", "activationBorderColor": "#8b949e"}}}%%
flowchart TD
    A[compile_task Mode B] --> B{task_source set and no inner_task?}
    B -- No --> E[build_harbor_dataset + register]
    B -- Yes --> C{VERO_SKIP_TASK_NAME_CHECK?}
    C -- Yes --> E
    C -- No --> D[_resolve_task_source_names]
    D --> F{names returned?}
    F -- None or empty set --> G[logger.warning: skipping check]
    G --> E
    F -- non-empty set --> H{unknown names in partition?}
    H -- No --> E
    H -- Yes --> I[raise ValueError with sample canonical names]
    D --> J{Running event loop?}
    J -- No --> K[asyncio.run coro]
    J -- Yes --> L[ThreadPoolExecutor asyncio.run in worker thread]
    K --> M[return set of names]
    L --> M
```

</a>
</details>

<sub>Reviews (2): Last reviewed commit: ["fix(harbor): partition-name validation s..."](https://github.com/scaleapi/vero/commit/e5e4d9cb766fe28ae1a1f45c132ba99b7b0ac94f) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=41610133)</sub>

### CODE REVIEW

File: vero/src/vero/harbor/build/compiler.py [hunk 1]
```diff
@@ -142,6 +142,55 @@ def _register(dataset, vero_home: Path, tmp: Path) -> str:
...
+        import asyncio
+
+        from harbor.models.job.config import DatasetConfig
+
+        name = task_source.split("@")[0]
+        cfgs = asyncio.run(DatasetConfig(name=name).get_task_configs())
+        return {c.name for c in cfgs}
+    except Exception as e:
+        logger.debug(f"task_source enumeration failed: {type(e).__name__}: {e}")
+        return None
```
[the remark this answers is not in the document]
Comment (shehabyasser-scale) [reply vero/src/vero/harbor/build/compiler.py:170]: Real gap: enumeration now detects a running loop and runs asyncio.run on its own loop in a worker thread, so async callers get real validation instead of a silent skip behind a misleading warning. Fixed in e5e4d9c.

File: vero/src/vero/harbor/build/compiler.py [hunk 2]
```diff
@@ -142,6 +142,55 @@ def _register(dataset, vero_home: Path, tmp: Path) -> str:
...
+        return
+    unknown = sorted({t for tasks in partition.values() for t in tasks} - names)
+    if unknown:
+        sample = sorted(names)[:3]
+        raise ValueError(
+            f"partition contains task name(s) not found in task_source "
+            f"'{task_source}': {unknown[:5]}. Task names must use harbor's "
+            f"canonical '<org>/<name>' form, e.g. {sample}. "
+            f"(Set VERO_SKIP_TASK_NAME_CHECK=1 to bypass.)"
+        )
```
[the remark this answers is not in the document]
Comment (shehabyasser-scale) [reply vero/src/vero/harbor/build/compiler.py:207]: Fixed in e5e4d9c: empty enumeration now warns-and-skips exactly like enumeration failure, instead of raising with the useless 'e.g. []' hint.

File: vero/tests/test_harbor_build.py [hunk 3]
```diff
@@ -167,3 +167,38 @@ def test_seed_documents_advisory_read_only(built):
...
+            {"train": ["org/task-a"], "validation": ["org/task-b"]}, "org/bench"
+        )  # no raise
+
+    def test_offline_enumeration_warns_and_continues(self, monkeypatch, caplog):
+        from vero.harbor.build import compiler
+        monkeypatch.delenv("VERO_SKIP_TASK_NAME_CHECK", raising=False)
+        monkeypatch.setattr(compiler, "_resolve_task_source_names", lambda ts: None)
+        with caplog.at_level("WARNING"):
+            compiler._validate_partition_names({"train": ["anything"]}, "org/bench")
+        assert any("skipping the check" in r.message for r in caplog.records)
```
[the remark this answers is not in the document]
Comment (shehabyasser-scale) [reply vero/tests/test_harbor_build.py]: Added test_skip_env_var_bypasses_check (plus a test for the empty-source branch). Fixed in e5e4d9c.
